src/Config.php

```diff
@@ -40,63 +35,30 @@ class Config implements ConfigInterface
         'shared'             => true,
     ];
 
-    /**
-     * @var array<string,array>
-     * @psalm-var ServiceManagerConfigurationType
-     */
-    protected $config = [
-        'abstract_factories' => [],
-        'aliases'            => [],
-        'delegators'         => [],
-        'factories'          => [],
-        'initializers'       => [],
-        'invokables'         => [],
-        'lazy_services'      => [],
-        'services'           => [],
-        'shared'             => [],
-    ];
+    /** @var ServiceManagerConfigurationType */
+    private array $config;
```

Comment:
Not to be tackled here, but we may (in future) benchmark the creation and usage of this object with a nested array vs declared properties.

Reply:
I mark this as deprecated, as the object makes no sense anymore. 
I also dont think that an object will be more performant than an assoc array.